pkg/utils/configs/config.go

```diff
@@ -52,8 +52,23 @@ func GetQdrExposedListeners(m *v1alpha1.Qdr) []v1alpha1.Listener {
 	return listeners
 }
 
-func SetQdrDefaults(m *v1alpha1.Qdr) bool {
+func SetQdrDefaults(m *v1alpha1.Qdr) (bool, bool) {
 	requestCert := false
+	updateDefaults := false
+
+	if m.Spec.DeploymentPlan.Size == 0 {
+		m.Spec.DeploymentPlan.Size = 1
+		updateDefaults = true
+	}
+	if m.Spec.DeploymentPlan.Role == "" {
+		m.Spec.DeploymentPlan.Role = v1alpha1.RouterRoleInterior
+		m.Spec.DeploymentPlan.Placement = v1alpha1.PlacementAny
```

Comment:
If role is not specified but placement is set to every, then the placement would get overridden here. Is that what we want? (I accept that interior and every may be an odd combination, but do we want to silently exclude it?)